From 2b91dd06f73a557a865a5e1200ce08e907b1c131 Mon Sep 17 00:00:00 2001
From: lapplislazuli <Leonhard.Applis@Protonmail.com>
Date: Tue, 29 Nov 2022 14:59:31 +0100
Subject: [PATCH] Added Tests for HasBugs

---
 src/ShellCheck/Analytics.hs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ShellCheck/Analytics.hs b/src/ShellCheck/Analytics.hs
index d48104e..83fd45c 100644
--- a/src/ShellCheck/Analytics.hs
+++ b/src/ShellCheck/Analytics.hs
@@ -545,6 +545,10 @@ prop_checkPipePitfalls15 = verifyNot checkPipePitfalls "foo | grep bar | wc -cmw
 prop_checkPipePitfalls16 = verifyNot checkPipePitfalls "foo | grep -r bar | wc -l"
 prop_checkPipePitfalls17 = verifyNot checkPipePitfalls "foo | grep -l bar | wc -l"
 prop_checkPipePitfalls18 = verifyNot checkPipePitfalls "foo | grep -L bar | wc -l"
+prop_checkPipePitfalls19 = verifyNot checkPipePitfalls "foo | grep -A2 bar | wc -l"
+prop_checkPipePitfalls20 = verifyNot checkPipePitfalls "foo | grep -B999 bar | wc -l"
+prop_checkPipePitfalls21 = verifyNot checkPipePitfalls "foo | grep --after-context 999 bar | wc -l"
+prop_checkPipePitfalls22 = verifyNot checkPipePitfalls "foo | grep -B 1 --after-context 999 bar | wc -l"
 checkPipePitfalls _ (T_Pipeline id _ commands) = do
     for ["find", "xargs"] $
         \(find:xargs:_) ->
-- 
2.37.1 (Apple Git-137.1)

