From 65d1b92d218a37f2d347023abc799a46efac0adc Mon Sep 17 00:00:00 2001
From: Leonhard Applis <Leonhard.Applis@Protonmail.com>
Date: Fri, 9 Sep 2022 15:32:18 +0200
Subject: [PATCH] Test for nested mapfiles reference

---
 src/ShellCheck/Analytics.hs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ShellCheck/Analytics.hs b/src/ShellCheck/Analytics.hs
index 582b8cd..d6ea460 100644
--- a/src/ShellCheck/Analytics.hs
+++ b/src/ShellCheck/Analytics.hs
@@ -2190,6 +2190,7 @@ prop_checkUnassignedReferences36= verifyNotTree checkUnassignedReferences "read
 prop_checkUnassignedReferences37= verifyNotTree checkUnassignedReferences "var=howdy; printf -v 'array[0]' %s \"$var\"; printf %s \"${array[0]}\";"
 prop_checkUnassignedReferences38= verifyTree (checkUnassignedReferences' True) "echo $VAR"
 prop_checkUnassignedReferences39= verifyNotTree checkUnassignedReferences "builtin export var=4; echo $var"
+prop_checkUnassignedReferences40= verifyNotTree checkUnassignedReferences "mapfile -t files <(cat); echo \"${files[@]}\""
 
 checkUnassignedReferences = checkUnassignedReferences' False
 checkUnassignedReferences' includeGlobals params t = warnings
-- 
2.24.3 (Apple Git-128)

