From 078c7bbc104c473b1b5b5fd5d071a1d79481c3ce Mon Sep 17 00:00:00 2001
From: lapplislazuli <Leonhard.Applis@Protonmail.com>
Date: Tue, 29 Nov 2022 15:26:15 +0100
Subject: [PATCH] Added Tests from 1302

---
 src/ShellCheck/Parser.hs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ShellCheck/Parser.hs b/src/ShellCheck/Parser.hs
index 1255d7e..5e7d82c 100644
--- a/src/ShellCheck/Parser.hs
+++ b/src/ShellCheck/Parser.hs
@@ -2499,6 +2499,7 @@ prop_readFunctionDefinition9 = isOk readFunctionDefinition "function foo { true;
 prop_readFunctionDefinition10= isOk readFunctionDefinition "function foo () { true; }"
 prop_readFunctionDefinition11= isWarning readFunctionDefinition "function foo{\ntrue\n}"
 prop_readFunctionDefinition12= isOk readFunctionDefinition "function []!() { true; }"
+prop_readFunctionDefinition13= isOk readFunctionDefinition "@require(){ true; }"
 readFunctionDefinition = called "function" $ do
     start <- startSpan
     functionSignature <- try readFunctionSignature
-- 
2.37.1 (Apple Git-137.1)

