From 1d260fb75adc7c2b09e4455ac6387ec4d5f3721b Mon Sep 17 00:00:00 2001
From: lapplislazuli <Leonhard.Applis@Protonmail.com>
Date: Fri, 25 Nov 2022 14:56:29 +0100
Subject: [PATCH] Added two properties

---
 src/ShellCheck/Analytics.hs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ShellCheck/Analytics.hs b/src/ShellCheck/Analytics.hs
index 88947a6..67077ca 100644
--- a/src/ShellCheck/Analytics.hs
+++ b/src/ShellCheck/Analytics.hs
@@ -2519,6 +2519,8 @@ prop_checkCharRangeGlob3 = verify checkCharRangeGlob "ls [10-15]"
 prop_checkCharRangeGlob4 = verifyNot checkCharRangeGlob "ls [a-zA-Z]"
 prop_checkCharRangeGlob5 = verifyNot checkCharRangeGlob "tr -d [a-zA-Z]" -- tr has 2060
 prop_checkCharRangeGlob6 = verifyNot checkCharRangeGlob "[[ $x == [!!]* ]]"
+prop_checkCharRangeGlob7 = verifyNot checkCharRangeGlob "[[ -v arr[keykey] ]]"
+prop_checkCharRangeGlob8 = verifyNot checkCharRangeGlob "[[ arr[keykey] -gt 1 ]]"
 checkCharRangeGlob p t@(T_Glob id str) |
   isCharClass str && not (isParamTo (parentMap p) "tr" t) =
     if ":" `isPrefixOf` contents
-- 
2.37.1 (Apple Git-137.1)

