From fbdbd2c7c4392cf873c67ee283d9b39e1f90a8e8 Mon Sep 17 00:00:00 2001
From: Leonhard Applis <Leonhard.Applis@Protonmail.com>
Date: Mon, 14 Jun 2021 14:06:28 +0200
Subject: [PATCH] Added a Test for Annulus KMeans in #87

---
 .../elki/clustering/kmeans/AnnulusKMeansTest.java    | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/elki-clustering/src/test/java/elki/clustering/kmeans/AnnulusKMeansTest.java b/elki-clustering/src/test/java/elki/clustering/kmeans/AnnulusKMeansTest.java
index 56326b16e..25598de08 100644
--- a/elki-clustering/src/test/java/elki/clustering/kmeans/AnnulusKMeansTest.java
+++ b/elki-clustering/src/test/java/elki/clustering/kmeans/AnnulusKMeansTest.java
@@ -45,4 +45,16 @@ public class AnnulusKMeansTest extends AbstractClusterAlgorithmTest {
     assertFMeasure(db, result, 0.998005);
     assertClusterSizes(result, new int[] { 199, 200, 200, 200, 201 });
   }
+
+  // Covers Issue 87
+  @Test
+  public void testKMeansAnnulus_SingleCluster_shouldPutAllInOneCluster() {
+    Database db = makeSimpleDatabase(UNITTEST + "different-densities-2d-no-noise.ascii", 1000);
+    Clustering<?> result = new ELKIBuilder<AnnulusKMeans<DoubleVector>>(AnnulusKMeans.class) //
+            .with(KMeans.K_ID, 1) //
+            .with(KMeans.SEED_ID, 7) //
+            .build().autorun(db);
+    assertClusterSizes(result, new int[] { 1000 });
+  }
+
 }
-- 
2.24.3 (Apple Git-128)

